### ISSUE TITLE
Error in Custom ZoomIn and ZoomOut Buttons

### ISSUE THREAD

**Issue by Andyeyo** [osmdroid/osmdroid#1319]
Please note: issues related to or caused by osmbonuspack will be closed with a
suggestion to open an issue at https://github.com/MKergall/osmbonuspack

(feel free to delete whatever doesn't apply)

## Issue Type

[ X ] Question
[ X ] Bug
[ ] Improvement
[ ] Build system related
[ ] Performance
[ ] Documentation

## Description and/or steps/code to reproduce the problem

I have added custom buttons for zoomin and zoomout, but when I clicked it, launch an error. My code is the following 
``` java
@Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater,
                             @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        context = container.getContext();

        // Check for update
        // UpdateChecker.checkForDialog( context);

        //handle permissions first, before map is created. not depicted here
        // Android 6.0+ devices require you have to check for "dangerous" permissions at runtime
        checkPermissions();

        // It is used to define user agent and navigate with credentials
        Configuration.getInstance().setUserAgentValue(BuildConfig.APPLICATION_ID);

        rootView = inflater.inflate(R.layout.fragment_home, null);

        initializeMap();

        return rootView;
    }


 private void initializeMap() {

        map = rootView.findViewById(R.id.map);
        map.setTileSource(TileSourceFactory.MAPNIK);

        btCenterMap = rootView.findViewById(R.id.ic_center_map);
        btCenterMap.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mLocationOverlay.enableMyLocation();
                mLocationOverlay.enableFollowLocation();
            }
        });
        
        btZoomInMap = rootView.findViewById(R.id.ic_zoom_in);
        btZoomInMap.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                map.getController().zoomIn();
                checkzoom();
            }
        });

        btZoomOutMap = rootView.findViewById(R.id.ic_zoom_out);
        btZoomOutMap.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                map.getController().zoomOut();
                checkzoom();
            }
        });
        }

    public void checkzoom() {
        if(map.getZoomLevel()==1) // dont know if 0 or 1, just wrote it
        {
            btZoomOutMap.setEnabled(false);
            // or u can change drawable and disable click
        }
        if(map.getZoomLevel()==18) {
            btZoomInMap.setEnabled(false);
            // or u can change drawable and disable click
        }
    }
``` 
When I clicked zoomin or zoomout button, I got the following error:

``` java

E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.example.yendoapp, PID: 8535
    java.lang.StackOverflowError: stack size 8MB
        at org.osmdroid.util.TileSystemWebMercator.getX01FromLongitude(TileSystemWebMercator.java:16)
        at org.osmdroid.util.TileSystem.getX01FromLongitude(TileSystem.java:221)
        at org.osmdroid.util.TileSystem.getMercatorXFromLongitude(TileSystem.java:489)
        at org.osmdroid.views.Projection.<init>(Projection.java:94)
        at org.osmdroid.views.Projection.<init>(Projection.java:65)
        at org.osmdroid.views.MapView.getProjection(MapView.java:358)
        at org.osmdroid.views.MapView.setExpectedCenter(MapView.java:1809)
        at org.osmdroid.views.MapView.setExpectedCenter(MapView.java:1823)
        at org.osmdroid.views.MapView.setZoomLevel(MapView.java:480)
        at org.osmdroid.views.MapController$MapAnimatorListener.onAnimationUpdate(MapController.java:520)
        at android.animation.ValueAnimator.animateValue(ValueAnimator.java:1374)
        at android.animation.ValueAnimator.end(ValueAnimator.java:1088)
        at org.osmdroid.views.MapController.animateTo(MapController.java:167)
        at org.osmdroid.views.MapController.animateTo(MapController.java:137)
        at org.osmdroid.views.MapController.animateTo(MapController.java:183)
        at org.osmdroid.views.MapController.animateTo(MapController.java:129)
        at org.osmdroid.views.MapView.setZoomLevel(MapView.java:492)
        at org.osmdroid.views.MapController$MapAnimatorListener.onAnimationUpdate(MapController.java:520)
        at android.animation.ValueAnimator.animateValue(ValueAnimator.java:1374)
        at android.animation.ValueAnimator.end(ValueAnimator.java:1088)
        at org.osmdroid.views.MapController.animateTo(MapController.java:167)
        at org.osmdroid.views.MapController.animateTo(MapController.java:137)
        at org.osmdroid.views.MapController.animateTo(MapController.java:183)
        at org.osmdroid.views.MapController.animateTo(MapController.java:129)
        at org.osmdroid.views.MapView.setZoomLevel(MapView.java:492)
        at org.osmdroid.views.MapController$MapAnimatorListener.onAnimationUpdate(MapController.java:520)
        at android.animation.ValueAnimator.animateValue(ValueAnimator.java:1374)
        at android.animation.ValueAnimator.end(ValueAnimator.java:1088)
        at org.osmdroid.views.MapController.animateTo(MapController.java:167)
        at org.osmdroid.views.MapController.animateTo(MapController.java:137)
        at org.osmdroid.views.MapController.animateTo(MapController.java:183)
        at org.osmdroid.views.MapController.animateTo(MapController.java:129)
        at org.osmdroid.views.MapView.setZoomLevel(MapView.java:492)
        at org.osmdroid.views.MapController$MapAnimatorListener.onAnimationUpdate(MapController.java:520)
        at android.animation.ValueAnimator.animateValue(ValueAnimator.java:1374)
        at android.animation.ValueAnimator.end(ValueAnimator.java:1088)
        at org.osmdroid.views.MapController.animateTo(MapController.java:167)
        at org.osmdroid.views.MapController.animateTo(MapController.java:137)
        at org.osmdroid.views.MapController.animateTo(MapController.java:183)
        at org.osmdroid.views.MapController.animateTo(MapController.java:129)
        at org.osmdroid.views.MapView.setZoomLevel(MapView.java:492)
        at org.osmdroid.views.MapController$MapAnimatorListener.onAnimationUpdate(MapController.java:520)
        at android.animation.ValueAnimator.animateValue(ValueAnimator.java:1374)
        at android.animation.ValueAnimator.end(ValueAnimator.java:1088)
        at org.osmdroid.views.MapController.animateTo(MapController.java:167)
        at org.osmdroid.views.MapController.animateTo(MapController.java:137)
        at org.osmdroid.views.MapController.animateTo(MapController.java:183)
        at org.osmdroid.views.MapController.animateTo(MapController.java:129)
        at org.osmdroid.views.MapView.setZoomLevel(MapView.java:492)
        at org.osmdroid.views.MapController$MapAnimatorListener.onAnimationUpdate(MapController.java:520)
        at android.animation.ValueAnimator.animateValue(ValueAnimator.java:1374)
        at android.animation.ValueAnimator.end(ValueAnimator.java:1088)
        at org.osmdroid.views.MapController.animateTo(MapController.java:167)
        at org.osmdroid.views.MapController.animateTo(MapController.java:137)
    	at org.osmdroid.vi
D/Error: ERR: exClass=java.lang.StackOverflowError
    ERR: exMsg=stack size 8MB
    ERR: file=TileSystemWebMercator.java
    ERR: class=org.osmdroid.util.TileSystemWebMercator
    ERR: method=getX01FromLongitude line=16
D/Error: ERR: stack=java.lang.StackOverflowError: stack size 8MB
        at org.osmdroid.util.TileSystemWebMercator.getX01FromLongitude(TileSystemWebMercator.java:16)
        at org.osmdroid.util.TileSystem.getX01FromLongitude(TileSystem.java:221)
        at org.osmdroid.util.TileSystem.getMercatorXFromLongitude(TileSystem.java:489)
        at org.osmdroid.views.Projection.<init>(Projection.java:94)
        at org.osmdroid.views.Projection.<init>(Projection.java:65)
        at org.osmdroid.views.MapView.getProjection(MapView.java:358)
        at org.osmdroid.views.MapView.setExpectedCenter(MapView.java:1809)
        at org.osmdroid.views.MapView.setExpectedCenter(MapView.java:1823)
        at org.osmdroid.views.MapView.setZoomLevel(MapView.java:480)
        at org.osmdroid.views.MapController$MapAnimatorListener.onAnimationUpdate(MapController.java:520)
        at android.animation.ValueAnimator.animateValue(ValueAnimator.java:1374)
        at android.animation.ValueAnimator.end(ValueAnimator.java:1088)
        at org.osmdroid.views.MapController.animateTo(MapController.java:167)
        at org.osmdroid.views.MapController.animateTo(MapController.java:137)
        at org.osmdroid.views.MapController.animateTo(MapController.java:183)
        at org.osmdroid.views.MapController.animateTo(MapController.java:129)
        at org.osmdroid.views.MapView.setZoomLevel(MapView.java:492)
        at org.osmdroid.views.MapController$MapAnimatorListener.onAnimationUpdate(MapController.java:520)
        at android.animation.ValueAnimator.animateValue(ValueAnimator.java:1374)
        at android.animation.ValueAnimator.end(ValueAnimator.java:1088)
        at org.osmdroid.views.MapController.animateTo(MapController.java:167)
        at org.osmdroid.views.MapController.animateTo(MapController.java:137)
        at org.osmdroid.views.MapController.animateTo(MapController.java:183)
        at org.osmdroid.views.MapController.animateTo(MapController.java:129)
        at org.osmdroid.views.MapView.setZoomLevel(MapView.java:492)
        at org.osmdroid.views.MapController$MapAnimatorListener.onAnimationUpdate(MapController.java:520)
        at android.animation.ValueAnimator.animateValue(ValueAnimator.java:1374)
        at android.animation.ValueAnimator.end(ValueAnimator.java:1088)
        at org.osmdroid.views.MapController.animateTo(MapController.java:167)
        at org.osmdroid.views.MapController.animateTo(MapController.java:137)
        at org.osmdroid.views.MapController.animateTo(MapController.java:183)
        at org.osmdroid.views.MapController.animateTo(MapController.java:129)
        at org.osmdroid.views.MapView.setZoomLevel(MapView.java:492)
        at org.osmdroid.views.MapController$MapAnimatorListener.onAnimationUpdate(MapController.java:520)
        at android.animation.ValueAnimator.animateValue(ValueAnimator.java:1374)
        at android.animation.ValueAnimator.end(ValueAnimator.java:1088)
        at org.osmdroid.views.MapController.animateTo(MapController.java:167)
        at org.osmdroid.views.MapController.animateTo(MapController.java:137)
        at org.osmdroid.views.MapController.animateTo(MapController.java:183)
        at org.osmdroid.views.MapController.animateTo(MapController.java:129)
        at org.osmdroid.views.MapView.setZoomLevel(MapView.java:492)
        at org.osmdroid.views.MapController$MapAnimatorListener.onAnimationUpdate(MapController.java:520)
        at android.animation.ValueAnimator.animateValue(ValueAnimator.java:1374)
        at android.animation.ValueAnimator.end(ValueAnimator.java:1088)
        at org.osmdroid.views.MapController.animateTo(MapController.java:167)
        at org.osmdroid.views.MapController.animateTo(MapController.java:137)
        at org.osmdroid.views.MapController.animateTo(MapController.java:183)
        at org.osmdroid.views.MapController.animateTo(MapController.java:129)
        at org.osmdroid.views.MapView.setZoomLevel(MapView.java:492)
        at org.osmdroid.views.MapController$MapAnimatorListener.onAnimationUpdate(MapController.java:520)
        at android.animation.ValueAnimator.animateValue(ValueAnimator.java:1374)
        at android.animation.ValueAnimator.end(ValueAnimator.java:1088)
        at org.osmdroid.views.MapController.animateTo(MapController.java:167)
        at org.osmdroid.views.MapController.animateTo(MapController.java:137)
        at org.osmdroid.views.MapController.animateTo(MapController.java:183)
    	at org.o
    ERR: TOTAL BYTES WRITTEN: 2174712
E/JavaBinder: !!! FAILED BINDER TRANSACTION !!!
E/AndroidRuntime: Error reporting crash
    android.os.TransactionTooLargeException
        at android.os.BinderProxy.transactNative(Native Method)
        at android.os.BinderProxy.transact(Binder.java:496)
        at android.app.ActivityManagerProxy.handleApplicationCrash(ActivityManagerNative.java:5192)
        at com.android.internal.os.RuntimeInit$UncaughtHandler.uncaughtException(RuntimeInit.java:95)
        at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:693)
        at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:690)
I/Process: Sending signal. PID: 8535 SIG: 9
Application terminated.
```
### If it's a bug, version(s) of android this affects:

All

### Version of osmdroid the issue relates to:

6.1.0

**Comment by JiKanghui** [comment]
Yes, I also have this problem.

**Comment by monsieurtanuki** [comment]
There are now "official" custom zoom buttons:
````java
mapView.getZoomController().setVisibility(CustomZoomButtonsController.Visibility.SHOW_AND_FADEOUT);
````

**Comment by mwuwer** [comment]
I'm facing the issue too after updating to 6.1. I'm using the osmbonuspack too. Seems to happen sporadically, more often right after the app has been started.

Is there an issue in https://github.com/MKergall/osmbonuspack that relates it (see note above)? Cannot find it actually.

**Comment by monsieurtanuki** [comment]
@mwuwer If it's at init time, it might be related to the infamous `post` syntax. For instance, in `onCreate`, actions related to the map centering/zooming must be embedded in a `mMapView.post(...)` syntax. Does that solve your issue?

**Comment by edy1** [comment]
The bug is still present! How do we fix it?

**Comment by monsieurtanuki** [comment]
@edy1 Still present in osmdroid 6.1.6 ? Do you have the same issues if you use the "official custom zoom buttons"?

**Comment by ikhsanhabibi** [comment]
I have the same problem and I use this:

 map.getZoomController().setVisibility(CustomZoomButtonsController.Visibility.ALWAYS);


Whenever I click follow me button, still got crash:

E/JavaBinder: !!! FAILED BINDER TRANSACTION !!!
E/AndroidRuntime: Error reporting crash
    android.os.TransactionTooLargeException

Please help!

**Comment by monsieurtanuki** [comment]
@ikhsanhabibi Is that a problem with the zoom buttons, or with the follow me feature?
If you have this "follow me" crash regardless of the zoom buttons, it's another topic: try to find another issue related to "follow me" where you could find the answer, or create a new issue.

**Comment by ikhsanhabibi** [comment]
@monsieurtanuki I guess, it's because of "follow me" button. Whenever I tried to click that button, I cannot zoom in or zoom out again, but at first it works before clicking that button. Alright 👌

**Comment by mwuwer** [comment]
@monsieurtanuki regarding your question if it's still present int 6.1.6 - it seems it does not! 
At least I upgraded to 6.1.6 and activated the zoom controls again and do not have any crashes now, but when I downgrade again to 6.1.0 it still happens.
So from my perspective it is fixed with 6.1.6

**Comment by monsieurtanuki** [comment]
@mwuwer Thank you for your test: I close the issue.

@Andyeyo @JiKanghui @edy1 Of course if you happen to see this issue with osmdroid 6.1.6, feel free to reopen this issue or to create a new one.
